File: python_sdk/src/reality_capture/specifications/eval_s2d.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum


class EvalS2DInputs(BaseModel):
    reference: str = Field(description="Reality data ID (cloud) or local path (on-premise) of ContextScene, "
                                       "pointing to segmented photos reference")
    prediction: str = Field(description="Reality data ID (cloud) or local path (on-premise) of ContextScene, "
                                        "pointing to segmented photos prediction")


class EvalS2DOutputs(BaseModel):
    report: Optional[str] = Field(None,
                                  description="Path to JSON report file with confusion matrix (bucket path in cloud, local path on-premise)")
    segmented_photos: Optional[str] = Field(None, alias="segmentedPhotos",
                                            description="Reality data ID (cloud) or local path (on-premise) of segmented photos, "
                                                        "annotated with confusion matrix index")
    segmentation2d: Optional[str] = Field(None, alias="segmentation2D",
                                          description="Reality data ID (cloud) or local path (on-premise) of ContextScene, "
                                                      "pointing to segmented photos")


class EvalS2DOutputsCreate(Enum):
    REPORT = "report"
    SEGMENTED_PHOTOS = "segmentedPhotos"
    SEGMENTATION2D = "segmentation2D"


class EvalS2DSpecificationsCreate(BaseModel):
    inputs: EvalS2DInputs = Field(description="Inputs")
    outputs: list[EvalS2DOutputsCreate] = Field(description="Outputs")


class EvalS2DSpecifications(BaseModel):
    inputs: EvalS2DInputs = Field(description="Inputs")
    outputs: EvalS2DOutputs = Field(description="Outputs")

File: typescript_sdk/reality_capture/src/specifications/eval_s2d.ts
import { z } from "zod";

export const EvalS2DInputsSchema = z.object({
  reference: z.string().describe("Reality data id of ContextScene, pointing to segmented photos reference"),
  prediction: z.string().describe("Reality data id of ContextScene, pointing to segmented photos prediction"),
});
export type EvalS2DInputs = z.infer<typeof EvalS2DInputsSchema>;

export const EvalS2DOutputsSchema = z.object({
  report: z.string()
    .regex(/^bkt:.+/)
    .describe("Path in Bucket of json report with confusion matrix")
    .optional(),
  segmentedPhotos: z.string()
    .describe("Reality data id of segmented photos, annotated with confusion matrix index")
    .optional(),
  segmentation2D: z.string()
    .describe("Reality data id of ContextScene, pointing to segmented photos")
    .optional(),
});
export type EvalS2DOutputs = z.infer<typeof EvalS2DOutputsSchema>;

export enum EvalS2DOutputsCreate {
  REPORT = "report",
  SEGMENTED_PHOTOS = "segmentedPhotos",
  SEGMENTATION2D = "segmentation2D"
}

export const EvalS2DSpecificationsCreateSchema = z.object({
  inputs: EvalS2DInputsSchema.describe("Inputs"),
  outputs: z.array(z.nativeEnum(EvalS2DOutputsCreate)).describe("Outputs"),
});
export type EvalS2DSpecificationsCreate = z.infer<typeof EvalS2DSpecificationsCreateSchema>;

export const EvalS2DSpecificationsSchema = z.object({
  inputs: EvalS2DInputsSchema.describe("Inputs"),
  outputs: EvalS2DOutputsSchema.describe("Outputs"),
});
export type EvalS2DSpecifications = z.infer<typeof EvalS2DSpecificationsSchema>;